btservice/AndroidBluetoothServer.cpp

```diff
@@ -103,115 +68,147 @@ void AndroidBluetoothServer::writeSocketInfoMessage()
     {
-        OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sent socket info.";
-        handshakeState_ = ConnectionStatus::SENT_SOCKETINFO_MESSAGE;
+        OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sent SocketInfoRequest.";
     }
     else
     {
-        OPENAUTO_LOG(error) << "[AndroidBluetoothServer] Error sending socket Info.";
-        handshakeState_ = ConnectionStatus::ERROR;
+        OPENAUTO_LOG(error) << "[AndroidBluetoothServer] Error sending SocketInfoRequest.";
     }
 }
+void AndroidBluetoothServer::writeSocketInfoResponse()
+{
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sending SocketInfoResponse.";
+    QString ipAddr;
+    foreach(QHostAddress addr, QNetworkInterface::allAddresses())
+    {
+        if(!addr.isLoopback() && (addr.protocol() == QAbstractSocket::IPv4Protocol))
+        {
+            ipAddr = addr.toString();
+        }
+    }
+
+    btservice::proto::SocketInfoResponse socketInfoResponse;
+    socketInfoResponse.set_ip_address(ipAddr.toStdString());
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] ipAddress: "<< ipAddr.toStdString();
+
+    socketInfoResponse.set_port(5000);
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] port: "<< 5000;
+
+    socketInfoResponse.set_status(btservice::proto::Status::STATUS_SUCCESS);
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] status: "<< btservice::proto::Status::STATUS_SUCCESS;
+
+
+    if(this->writeProtoMessage(7, socketInfoResponse))
+    {
+        OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sent SocketInfoResponse.";
+    }
+    else
+    {
+        OPENAUTO_LOG(error) << "[AndroidBluetoothServer] Error sending SocketInfoResponse.";
+    }
+}
+
+void AndroidBluetoothServer::handleSocketInfoRequestResponse(QByteArray data)
+{
+    btservice::proto::SocketInfoResponse socketInfoResponse;
+    socketInfoResponse.ParseFromArray(data, data.size());
+    OPENAUTO_LOG(info) <<"[AndroidBluetoothServer] Received SocketInfoRequestResponse, status: "<<socketInfoResponse.status();
+}
+
+
+void AndroidBluetoothServer::handleSocketInfoRequest(QByteArray data)
+{
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Reading SocketInfoRequest.";
+
+    btservice::proto::SocketInfoRequest socketInfoRequest;
+    
+    writeSocketInfoResponse();
+}
 
 void AndroidBluetoothServer::writeNetworkInfoMessage()
 {
-    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sending network packet.";
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] Sending NetworkInfoMessage.";
 
     btservice::proto::NetworkInfo networkMessage;
     networkMessage.set_ssid(config_->getWifiSSID());
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] SSID: "<<config_->getWifiSSID();
+
     networkMessage.set_psk(config_->getWifiPassword());
-    foreach(QNetworkInterface netInterface, QNetworkInterface::allInterfaces())
+    OPENAUTO_LOG(info) << "[AndroidBluetoothServer] PSKEY: "<<config_->getWifiPassword();
+
+    if(config_->getWifiMAC().empty())
```

Comment:
in the future, it seems easy enough to rework this to have a UI element that allows the user to set which interface they want to use right? Like wlan0, wlan1, ...

looks like QNetworkInterface has a `allInterfaces` function that can get us all available interfaces

is there a reason you went with having the user type in a MAC address vs the interface name?

Reply:
Yes would be easy to rework interface wise, but on most use cases (pi) there will just be wlan0.

> is there a reason you went with having the user type in a MAC address vs the interface name?
Assuming you're talking about the openauto config value? The config value only needs to be set if you're using an external hotspot  or network (not the raspberry pi), if you're using the pi as a hotspot, it can be left blank.

Or in other words - for the scenarios in which you'd manually enter a MAC address, the pi is separate from the interface hosting the network.

Reply:
oh i see when i saw external network for some reason i assumed like a wifi dongle or something :stuck_out_tongue_closed_eyes: 

yeah im just really trying to come up with a way for a user to set that (with no keyboard) without needing to manually edit the config file... getting the MAC address of an external device tho... is challenging to say the least lol

Reply:
Ah - yea.

I wouldn't consider this a typical use case at least. The majority of users shouldn't need to touch the MAC. Although setting of ssid/pskey will still need to be done.